web/docs/internal/keystroke-lifecycle.md

```diff
@@ -0,0 +1,90 @@
+
+### `app/webview`
+
+The `app/webview` version of the engine, which is designed to be used while embedded in a platform-specific host app, does not have direct access to standard hardware keystroke events.  It's run in a WebView that does not take input - a WebView owned by the host app.  It is the responsibility of the host app to handle hardware keystroke events and preprocess them on behalf of the Web engine, then forward them to the Web engine via a carefully-constructed JS call from the host upon the WebView.  The `PassthroughKeyboard` class within `app/webview` space provides the method `raiseKeyEvent` as an internal API for this purpose, which converts the mobile-app format for hardware keystroke into the internal `KeyEvent` format.  This includes mnemonic keystroke processing and remapping.
+
+### On-screen keyboard
+
+For both `app/browser` and `app/webview`, the on-screen keyboard delegates mouse and touch processing to [our internal gesture engine](./gestures/index.md) for most keyboards.  (Keyboards that instead display "help pages", such as `sil_euro_latin` on desktop, have specialized handling.)  The gesture engine is responsible for interpreting all DOM mouse and touch events for OSKs based on the touch-layout format, which are managed by the `VisualKeyboard` class.  The gesture engine will raise events about the gesture lifecycle that, when appropriate, will emit `KeyElement` instances - elements of the constructed OSK corresponding to elements of the OSK directly representing keyboard keys.
+
+OSK keystroke preprocessing operates a little differently from hardware keystrokes.  The `ActiveLayout` object for the current form-factor of the `Keyboard` maintains a "base" `KeyEvent` object, per key, that is lazy-constructed (by the `.baseKeyEvent` property) when first needed - the meat of which is performed by the `ActiveKeyBase.constructBaseKeyEvent` method.  This is only evaluated up to once per key per activation of the keyboard.  As keystrokes are received during OSK use, `VisualKeyboard.keyEventFromSpec` calls the `Keyboard.constructKeyEvent` method to clone that base `KeyEvent` object for the incoming keystroke and then adjust it to match current modifier and state-key settings.
+
+### Final steps
+
+Regardless of the path taken to generate the `KeyEvent` object, _all_ keystroke handling modules of Keyman Engine for Web then raise that `KeyEvent` object via their `keyevent` event as specified by the `KeyEventSourceInterface` type.  This event also permits a callback that is used by the final phase of the lifecycle - keystroke resolution - as needed.  The main `KeymanEngine` base class listens for these events and forwards them for next phase of the lifecycle - keystroke processing - to the `InputProcessor` module, which passes control to whichever `KeystrokeProcessor` type is active for the keyboard when appropriate.
```

Comment:
I think an overview (and flow diagram) of the different components involved, before the "Keystroke pre-processing" section may help. For example, here you say "whichever `KeystrokeProcessor` type" but the concept of KeystrokeProcessor has not yet been introduced so the reader now has a question about what are KeystrokeProcessors and why would there be more than one (and then how/why are they selected)?